log/src/main/java/kcl/teamIndexZero/traffic/log/FileIO/FileIO.java

```diff
@@ -52,69 +59,90 @@ public boolean createFile() {
     /**
      * Deletes the file if it exists
      * return Success
-     * @exception InvalidPathException when the path description is invalid
-     * @exception IOException when the file cannot be accessed
+     *
+     * @throws InvalidPathException when the path description is invalid
+     * @throws IOException          when the file cannot be accessed
      */
-    public boolean deleteFile() throws InvalidPathException, IOException {
+    public synchronized boolean deleteFile() throws InvalidPathException, IOException {
         try {
-            if( Files.exists( this.getFilePath() )) {
-                Files.delete( this.getFilePath() );
+            if (Files.exists(this.getFilePath())) {
+                Files.delete(this.getFilePath());
```

Comment:
Again, a bit risky here. What if we had valuable logs in there? I.e. I'm just running an app and it deletes the file which was there before - a bit risky behaviour. What if we instead created .1 .2 etc. files - numbered ones?